core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -3448,6 +3448,13 @@ public static String createName( String filename ) {
     }
 
     String pureFilename = filenameOnly( filename );
+    if ( filename.startsWith( "pvfs://" ) ) {
+      try {
+        pureFilename = UriParser.decode( pureFilename );
+      } catch ( FileSystemException e ) {
+        // use as-is
+      }
+    }
```

Comment:
Offline feedback has suggested to use KettleVFS.resolveURI for this, so I will try that.